### DISCUSSION TITLE
Viewer

### DISCUSSION THREAD

**Discussion by stromhaug-tommy** [Macad3D/Macad3D#30] [General]
Hi,

I am trying to make a standalone simple viewer, When looking at your source-code i am not able to figure out how you connect your viewer object to your main window. In c++ its done with this : myView()->SetWindow(aWNTWindow);
The V3d_View object has a setWindow() that takes a aspect_window.
Could you please help me ?

Tommy

**Comment by dipts** [comment]
Hi Tommy,

the window ist created in [ViewportController,InitWindow](https://github.com/Macad3D/Macad3D/blob/main/Source/Macad.Interaction/Workspace/ViewportController.cs#L148), then it is embeded into a [HwndHost-Control](https://github.com/Macad3D/Macad3D/blob/main/Source/Macad.Interaction/Panels/Viewport/ViewportHwndHost.cs).

I hope this helps you.

Tobias

**Reply by stromhaug-tommy** [reply]
Thank you very much!
